### PR TITLE
Update shopcart

### PR DESCRIPTION
**Pull request by jules203**
# Pull Request Template

## Checklist:

- [x] I have written test cases for this change, and my code passed the test
- [x] I have updated documentation to reflect this change
- [x] I have attached this PR to an issue/user story

### CODE REVIEW

File: tests/test_service_routes.py [hunk 1]
```diff
@@ -257,6 +257,28 @@ def test_create_shopcart_item(self):
         #    new_shopcart_item["update_time"], "Update time not set"
         #)
 
+    def test_update_shopcart_item(self):
+        """ Update an existing shopcart item """
+        # create a shopcart item to update
+        test_shopcart_item = ShopcartFactory()
```
Comment (KHafner) [inline tests/test_service_routes.py:263]: ShopcartFactory should be ShopcartItemFactory

File: tests/test_service_routes.py [hunk 2]
```diff
@@ -331,6 +331,27 @@ def test_create_shopcart_item(self):
...
+        # create a shopcart item to update
+        test_shopcart_item = ShopcartItemFactory()
+        resp = self.app.post(
+            "/shopcartitems", json=test_shopcart_item.serialize(), content_type="application/json"
+        )
+        self.assertEqual(resp.status_code, status.HTTP_201_CREATED)
+
+        # update the shopcart item
+        new_shopcart_item = resp.get_json()
+        new_shopcart_item["price"] = "50.00"
```
Comment (KHafner) [inline tests/test_service_routes.py:345]: Add the lines here:
        new_shopcart_item["sku"] = 1001
        new_shopcart_item["name"] = item_1
        new_shopcart_item["amount"] = 4

File: tests/test_service_routes.py [hunk 3]
```diff
@@ -331,6 +331,27 @@ def test_create_shopcart_item(self):
...
+        resp = self.app.put(
+            "/shopcartitems/{}".format(new_shopcart_item["id"]),
+            json=new_shopcart_item,
+            content_type="application/json",
+        )
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        updated_shopcart_item = resp.get_json()
+        self.assertEqual(updated_shopcart_item["category"], "unknown")
```
Comment (KHafner) [inline tests/test_service_routes.py:353]: and add here:

self.assertEqual(updated_shopcart_item["sku"], 1001)
self.assertEqual(updated_shopcart_item["name"], "item_1")
self.assertEqual(updated_shopcart_item["amount"], 4)

File: tests/test_models.py [hunk 4]
```diff
@@ -220,6 +220,23 @@ def test_create_a_shopcart_item(self):
...
 
+    def test_update_a_shopcart_item(self):
+        """ Update a shopcart item """
+        date_time = datetime.now()
+        shopcart_item = ShopcartItem(sid=100, sku=5000, name="soap", price=2.23,
+                                    amount=3, create_time=date_time, update_time=date_time)
+        shopcart_item.create()
+        self.assertEqual(shopcart_item.id, 1)
+        # Change it an update it
+        shopcart_item.category = "soap"
```
Comment (KHafner) [inline tests/test_models.py:231]: There's no category member in ShopcartItem, this should be name

File: tests/test_models.py [hunk 5]
```diff
@@ -220,6 +220,23 @@ def test_create_a_shopcart_item(self):
...
+        shopcart_item.update()
+        self.assertEqual(shopcart_item.id, 1)
+        # Fetch it back and make sure the id hasn't changed
+        # but the data did change
+        shopcart_item = ShopcartItem.all()
+        self.assertEqual(len(shopcart_item), 1)
+        self.assertEqual(shopcart_item[0].category, "soap")
```
Comment (KHafner) [inline tests/test_models.py:238]: Here as well should be name not category

File: tests/test_service_routes.py [hunk 6]
```diff
@@ -331,6 +331,30 @@ def test_create_shopcart_item(self):
...
+        new_shopcart_item["name"] = "item_1"
+        new_shopcart_item["amount"] = 4
+        resp = self.app.put(
+            "/shopcartitems/{}".format(new_shopcart_item["id"]),
+            json=new_shopcart_item,
+            content_type="application/json",
+        )
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        updated_shopcart_item = resp.get_json()
+        self.assertEqual(updated_shopcart_item["price"], 50.00)
```
Comment (KHafner) [inline tests/test_service_routes.py:356]: can you add the lines:
self.assertEqual(updated_shopcart_item["sku"], 1001)
self.assertEqual(updated_shopcart_item["name"], "item_1")
self.assertEqual(updated_shopcart_item["amount"], 4)

File: README.md [hunk 7]
```diff
@@ -177,10 +177,14 @@ A shopcart list can be got with a `GET` request on `'/shopcarts'`. The response
     }
 ]
 ```
+## Updating a Shopcart Item
+A shopcart item can be updated with a `PUT` request on `'/shopcartitems'`.
```
Comment (KHafner) [inline README.md:181]: Should be /shopcartitems/id 
And include that of the field(s) to be included in the content for the PUT request, such as:
```json
{
    "amount": 4,
}
```

**Review by iLtc — APPROVED** [review]
LGTM, thanks!

File: README.md [hunk 8]
```diff
@@ -177,10 +177,20 @@ A shopcart list can be got with a `GET` request on `'/shopcarts'`. The response
...
+## Updating a Shopcart Item
+A shopcart item can be updated with a `PUT` request on `'/shopcartitems/id'`.
+
+When an authorized user hits a PUT request on '/shopcartitems', the API will return the updated shopcart item with a status code 200. The response will be something like:
+
+```json
+{
+    "amount": 4,
+}
+```
```
Comment (KHafner) [inline README.md:189]: Would you mind clarifying that this JSON would be the content for the `PUT` request, not the response. The response contains the full shopcart item

Comment (jules203) [reply README.md:189]: done!

**Review by KHafner — APPROVED** [review]
Thanks, looks good!

**Pull request merged**